operator-pipeline-images/operatorcert/static_tests/community/bundle.py

```diff
@@ -34,10 +34,53 @@ class GraphLoopException(Exception):
     """
 
 
+def extract_ocp_version_from_bundle_metadata(ocp_metadata_version: Any) -> Any:
```

Comment:
The `com.redhat.openshift.versions` can be defined in more formats than just a simple version or range. Please make sure all possible formats are supported or get the supported version form Pyxis based on the annotation.

Reply:
I grepped the community and certified repos for metadata annotations and found out that equal sign is also used to annotate single ocp version, so I included additional check here.

Reply:
@tomasbakk The version definition in this field is more than just a static selection of a single version. As a user, you can define multiple ways and each version format has a different meaning. That's why I suggest using a Pyxis to determine the latest supported version based on the annotation.

For example when a version annotation defines `v4.10` it means it supports all versions from `4.10` up to the latest supported one which is currently `4.14`.

Here is the query that gives you a supported version based on the annotation. From that list, you will use the last version available.
https://catalog.redhat.com/api/containers/v1/operators/indices?filter=organization==certified-operators&ocp_versions_range=v4.10

If you wan to read more about this annotation here is a detailed documentation: https://docs.engineering.redhat.com/display/CFC/Delivery

Reply:
aha, thank you! i thought that users have more options how to annotate the version, but in the end what matters is the upper ocp range (if range is mentioned) or single definition. now that you mentioned it, i remember that we discussed the supported versions during some meeting past week, and it makes sense to me now. i will go through the documentation and fix it. thanks.

Reply:
@Allda I added the call to get the indices and parse the OCP version based on that. Can you please check it out if it is alright? Thanks!